Gems/SurfaceData/Code/Source/TerrainSurfaceDataSystemComponent.cpp

```diff
@@ -168,12 +152,13 @@ namespace SurfaceData
                     const float terrainHeight = terrain->GetHeight(inPosition, AzFramework::Terrain::TerrainDataRequests::Sampler::BILINEAR, &isTerrainValidAtPoint);
                     const bool isHole = !isTerrainValidAtPoint;
 
-                    SurfacePoint point;
+                    SurfaceData::SurfacePoint point;
                     point.m_entityId = GetEntityId();
                     point.m_position = AZ::Vector3(inPosition.GetX(), inPosition.GetY(), terrainHeight);
                     point.m_normal = terrain->GetNormal(inPosition);
-                    const AZ::Crc32 terrainTag = isHole ? Constants::s_terrainHoleTagCrc : Constants::s_terrainTagCrc;
-                    AddMaxValueForMasks(point.m_masks, terrainTag, 1.0f);
+                    const AZ::Crc32 terrainTag =
+                        isHole ? SurfaceData::Constants::s_terrainHoleTagCrc : SurfaceData::Constants::s_terrainTagCrc;
```

Comment:
Curious why crcs for this. Seems like flags would be more useful.

Reply:
The SurfaceData system lets you specify any number of user-defined tags which are then used to tag surfaces, where the sets of tags are defined in surface tag assets.  There used to be several hard-coded default tags in the engine as well, but at this point, I think "terrain" and "terrainHole" are the only ones that remain.